src/core/sock/sockinfo_tcp.cpp

```diff
@@ -6061,13 +6234,63 @@ int sockinfo_tcp::tcp_tx_express(const struct iovec *iov, unsigned iov_len, uint
     }
     if (!(flags & XLIO_EXPRESS_MSG_MORE)) {
         tcp_output(&m_pcb);
+        m_b_xlio_socket_dirty = false;
+    } else if (m_p_group && !m_b_xlio_socket_dirty) {
```

Comment:
Should we check if dirty handling is requested?

Reply:
Currently it's checked inside the poll_group method, because this is the group configuration whether to track dirty sockets or not.